OpenKeychain/src/main/java/org/sufficientlysecure/keychain/ui/ImportKeysActivity.java

```diff
@@ -254,21 +271,26 @@ protected void onRestoreInstanceState(Bundle savedInstanceState) {
     }
 
     /**
-     * if the fragment is started with non-null bytes/dataUri/serverQuery, it will immediately
-     * load content
+     * Shows the list of keys to be imported.
+     * If the fragment is started with non-null bytes/dataUri/serverQuery, it will immediately
+     * load content.
      *
-     * @param bytes              bytes containing list of keyrings to import
-     * @param dataUri            uri to file to import keyrings from
-     * @param serverQuery        query to search for on the keyserver
-     * @param keyserver          keyserver authority to search on. If null will use keyserver from
-     *                           user preferences
+     * @param bytes            bytes containing list of keyrings to import
+     * @param dataUri          uri to file to import keyrings from
+     * @param serverQuery      query to search for on the keyserver
+     * @param cloudSearchPrefs search specifications to use. If null will retrieve from user's
+     *                         preferences.
      */
-    private void startListFragment(byte[] bytes, Uri dataUri, String serverQuery, String keyserver) {
+    private void startListFragment(byte[] bytes, Uri dataUri, String serverQuery,
+                                   Preferences.CloudSearchPrefs cloudSearchPrefs) {
         Fragment listFragment =
-                ImportKeysListFragment.newInstance(bytes, dataUri, serverQuery, false, keyserver);
+                ImportKeysListFragment.newInstance(bytes, dataUri, serverQuery, false,
+                        cloudSearchPrefs);
         getSupportFragmentManager().beginTransaction()
                 .replace(R.id.import_keys_list_container, listFragment, TAG_FRAG_LIST)
-                .commit();
+                .commitAllowingStateLoss();
```

Comment:
I'm really not a fan of `commitAllowingStateLoss`, and the general android advice is to use it as a last resort only. see above on handling `onNewIntent`.